src/main/java/frc/robot/TestMode.java

```diff
@@ -0,0 +1,239 @@
+        .finallyDo(algaeSubsystem::stop);
+  }
+
+  private Command testArmElevatorCommand() {
+    return run(() -> armSubsystem.moveElevatorLevel4(), armSubsystem).until(armSubsystem::isElevatorAtPosition)
+        .withTimeout(Seconds.of(5))
+        .andThen(this::updateElevatorTestResult)
+        .finallyDo(armSubsystem::moveElevatorToDefault);
+  }
+
+  private Command testArmCommand() {
+    return run(() -> armSubsystem.moveArmToLevel4(), armSubsystem).until(armSubsystem::isArmAtPosition)
+        .withTimeout(Seconds.of(5))
+        .andThen(this::updateArmTestResults)
+        .finallyDo(armSubsystem::moveArmToIntake);
+  }
+
+  private void updateIndexerForwardsTestResult() {
+    indexerForwardsTest = indexerSubsystem.isIndexerAtSpeed();
+  }
+
+  private void updateIndexerBackwardsTestResult() {
+    indexerBackwardsTest = indexerSubsystem.isIndexerAtSpeed();
+  }
+
+  private void updateManipulatorForwardsTestResult() {
+    manipulatorForwardsTest = gamePieceManipulatorSubsystem.isManipulatorAtSpeed();
+  }
+
+  private void updateManipulatorBackwardsTestResult() {
+    manipulatorBackwardsTest = gamePieceManipulatorSubsystem.isManipulatorAtSpeed();
+  }
+
+  private void updateClimbTestResult() {
+    climbTest = climbSubsystem.areClimbMotorsMoving();
+  }
+
+  private void updateAlgaeRollersForwardsTestResult() {
+    algaeRollersForwardsTest = algaeSubsystem.isAtRollerSpeed();
+  }
+
+  private void updateAlgaeRollersBackwardsTestResult() {
+    algaeRollersBackwardsTest = algaeSubsystem.isAtRollerSpeed();
+  }
+
+  private void updateAlgaeIntakeUpTestResult() {
+    algaeIntakeUpTest = algaeSubsystem.isWristAtPosition();
+  }
+
+  private void updateAlgaeIntakeDownTestResult() {
+    algaeIntakeDownTest = algaeSubsystem.isWristAtPosition();
+  }
+
+  private void updateElevatorTestResult() {
+    armElevatorTest = armSubsystem.isElevatorAtPosition();
+  }
+
+  private void updateArmTestResults() {
+    armTest = armSubsystem.isArmAtPosition();
+  }
+
+  public boolean getIndexerForwardsTestResult() {
+    return indexerForwardsTest;
+  }
+
+  public boolean getIndexerBackwardsTestResult() {
+    return indexerBackwardsTest;
+  }
+
+  public boolean getManipulatorForwardsTestResult() {
+    return manipulatorForwardsTest;
+  }
+
+  public boolean getManipulatorBackwardsTestResult() {
+    return manipulatorBackwardsTest;
+  }
+
+  public boolean getClimbTestResult() {
+    return climbTest;
+  }
+
+  public boolean getAlgaeRollersForwardsTestResult() {
+    return algaeRollersForwardsTest;
+  }
+
+  public boolean getAlgaeRollersBackwardsTestResult() {
+    return algaeRollersBackwardsTest;
+  }
+
+  public boolean getAlgaeIntakeUpTestResult() {
+    return algaeIntakeUpTest;
+  }
+
+  public boolean getAlgaeIntakeDownTestResult() {
+    return algaeIntakeDownTest;
+  }
+
+  public boolean getArmElevatorTestResult() {
+    return armElevatorTest;
+  }
+
+  public boolean getArmTestResult() {
+    return armTest;
+  }
```

Comment:
If we're following our best practice, these should have a JavaDoc comment that describes them. Ideally, the class Javadoc explains how the class can be used, and these have good descriptions of what they indicate.
Remember, the comments help us, but we also know that other teams look at our code for examples, and it helps them understand it.